test/e2e/httpproxy/002_header_condition_match_test.go

```diff
@@ -0,0 +1,302 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build e2e
+
+package httpproxy
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	contourv1 "github.com/projectcontour/contour/apis/projectcontour/v1"
+	"github.com/projectcontour/contour/test/e2e"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func testHeaderConditionMatch(t *testing.T, fx *e2e.Framework) {
+	namespace := "002-header-condition-match"
+
+	fx.CreateNamespace(namespace)
+	defer fx.DeleteNamespace(namespace)
+
+	fx.Fixtures.Echo.Create(namespace, "echo-header-present")
```

Comment:
just thinking out loud, maybe the abstraction could be that the "framework" orchestrates running the sub tests, but passes each one a "test case" type that it creates w/ the test name as the test namespace. it would have the fixture helpers fields etc. and that would already know to create everything in the test-name namespace already? we might want some facility to still override and create things in different namespaces for some tests so that might still be needed

Reply:
not necessarily changes to make to this PR but maybe just ideas to think about